pkg/orchestration/wiring/ec2compute/v2/plugin.go

```diff
@@ -159,7 +194,27 @@ func constructComputeParameters(origSpec *runtime.RawExtension, iamRoleRef, iamI
 	return rawExtension, false, false, nil
 }
 
-func WireUp(stateResource *orch_v1.StateResource, context *wiringplugin.WiringContext) wiringplugin.WiringResult {
+func New() *WiringPlugin {
+	return &WiringPlugin{
+		DeveloperRole: func(_ voyager.Location) []string {
+			return []string{"arn:aws:iam::123456789012:role/micros-server-iam-MicrosServer-ABC"} //example
```

Comment:
Legacy config is a _major_ issue for manual or experimental testing of autowiring functions. You have to use the internal build to access this data. I can probably move most or all of these to envvar lookups. (the way shapes hold things together also requires all the functions be in this repo)

Reply:
I don't disagree, it would be nice to replace LegacyConfig with resource-specific configuration.
The problem is, as I said above, that at the moment there is no way to pass configuration to autowiring plugins, and at the moment we are abusing WiringContext and LegacyConfig as a global per-environment configuration.

Reply:
There is a ticket to make autowiring functions accept configuration into their constructor. Even with legacy config completely gone we'd still need this functionality.

Reply:
That's exactly what I meant, if it wasn't clear :)

Reply:
We build the list of plugins. All of their "constructors" are accessible. There is not requirement to use known plugins object.

Reply:
Then the first trivial change would be to pass LegacyConfig to plugin constructors which use information from it, and remove it from the context? This would be a global LegacyConfig I suppose (not location-specific), but that's ok. It could also be a function providing a LegacyConfig for a given location. This should be a relatively simple isolated change, easy to review.

Reply:
Actually, we already have this function: https://github.com/atlassian/voyager/blob/master/pkg/orchestration/wiring/entangler.go#L161-L173
So we just need to pass this function to plugin constructors, instead of Entangler invoking it and adding to the context. The only problem again is that list of plugins is static at the moment IIRC, i.e. you can't pass a different configuration to it for testing vs production.

Reply:
Look at the main function for orchestration. It takes registry.KnownWiringPlugins as an arg of type map[voyager.ResourceType]wiringplugin.WiringPlugin. There is no reason this needs to be the same map inside the close source repo.

Reply:
Also notice that master of this repo currently stubs legacy config with struct that will nil pointer panic for anything that attempts to use it.